src/main/kotlin/OSSLLMCompletionProvider.kt

```diff
@@ -1,14 +1,28 @@
 package com.github.bzz.intellij
 
+import com.github.bzz.intellij.requests.Requester
 import com.intellij.codeInsight.inline.completion.InlineCompletionElement
-import com.intellij.codeInsight.inline.completion.InlineCompletionProvider;
+import com.intellij.codeInsight.inline.completion.InlineCompletionProvider
 import com.intellij.codeInsight.inline.completion.InlineCompletionRequest
+import com.intellij.openapi.diagnostic.thisLogger
 import com.intellij.openapi.editor.event.DocumentEvent
 
 class OSSLLMCompletionProvider : InlineCompletionProvider {
 
     override suspend fun getProposals(request: InlineCompletionRequest): List<InlineCompletionElement> {
-        return listOf(InlineCompletionElement("public static void"),InlineCompletionElement("this is a test \uD83E\uDD16"),)
+        try {
+            val caretPosition = request.startOffset
+            val proposal = Requester.getModelSuggestions(
+                request.document.text.substring(0, caretPosition)
```

Comment:
This probably should have an upper limit of, say 16/32/64k so we don't try to send 100Mb over the HTTP if the user opens a 100Mb file.

This can be added later on though.